src/components/Incidents/IncidentsPage.jsx

```diff
@@ -108,6 +113,10 @@ const IncidentsPage = ({ customDataSource, namespace }) => {
     })();
   }, [span]);
 
+  React.useEffect(() => {
+    setalertValuePairs(collectSeverityAndAlertnames(incidentsData));
```

Comment:
Should be this called when the incidents finished loading?

Reply:
Yes, I updated this in the latest commit.
So the scenario is following - We request the incidents, get "src_alertname" and "scr_severity" from them, then create custom unique pairs of values and then make new query that will look like
ALERTS{severity="smth",alertname="smth"} or ALERTS etc. 
That will be done for each custom combination of values. This is how we can request Alerts that match the Incidents.